File: lab-2/src/moves/SteelWing.java
package moves;

import ru.ifmo.se.pokemon.*;

public class SteelWing extends PhysicalMove {
    public SteelWing(){
        super(Type.STEEL, 70, 90);
    }

    protected void applySelfEffects(Pokemon p){
        if(Math.random()<0.1){
            p.setMod(Stat.DEFENSE, 1);
        }
    }

    protected String describe() {
        return "is using Steel Wing";
    }
}

File: lab-2-kotlin/src/moves/SteelWing.kt
package moves

import ru.ifmo.se.pokemon.*

class SteelWing : PhysicalMove(Type.STEEL, 70.0, 90.0) {
    override fun applySelfEffects(p: Pokemon) {
        if (Math.random() < 0.1) {
            p.setMod(Stat.DEFENSE, 1)
        }
    }

    override fun describe(): String {
        return "is using Steel Wing"
    }
}